crosshair/condition_parser.py

```diff
@@ -1161,12 +1168,106 @@ def get_class_invariants(self, cls: type) -> List[ConditionExpr]:
         return []
 
 
+class RegisteredContractsParser(ConcreteConditionParser):
+    """Parser for manually registered contracts."""
+
+    def __init__(self, toplevel_parser: ConditionParser = None):
+        super().__init__(toplevel_parser)
+
+    def get_fn_conditions(self, ctxfn: FunctionInfo) -> Optional[Conditions]:
+        fn_and_sig = ctxfn.get_callable()
+        if fn_and_sig is not None:
+            (fn, sig) = fn_and_sig
+            contract = get_contract(fn)
+            if not contract:
+                return None
+        else:
+            desc = ctxfn.descriptor
+            if isinstance(desc, Callable):  # type: ignore
+                fn = cast(Callable, desc)
+                contract = get_contract(fn)
+                if not contract or not contract.sig:
+                    return None
+                sig = contract.sig
+            else:
+                return None
+
+        pre: List[ConditionExpr] = []
+        post: List[ConditionExpr] = []
+
+        filename, line_num, _lines = sourcelines(fn)
+
+        if contract.pre:
+            pre_cond = contract.pre
+
+            def evaluatefn(kwargs: Mapping):
+                kwargs = dict(kwargs)
+                pre_args = inspect.signature(pre_cond).parameters.keys()
+                new_kwargs = {arg: kwargs[arg] for arg in pre_args}
+                return pre_cond(**new_kwargs)
+
+            pre.append(
+                ConditionExpr(
+                    PRECONDITION,
+                    evaluatefn,
+                    filename,
+                    line_num,
+                    f"manually registered precondition for {fn.__name__}",
+                )
+            )
+        if contract.post:
+            post_cond = contract.post
+
+            def post_eval(orig_kwargs: Mapping) -> bool:
+                kwargs = dict(orig_kwargs)
+                kwargs["result"] = kwargs["__return__"]
+                kwargs["OLD"] = kwargs["__old__"]
```

Comment:
I had the same reasoning last week. I think I will roll back to using `__return__` and `__old__`. There is no real reason of using `iContract`'s syntax more than PEP 316. Do you think we should also allow `_` in addition to `__return__` for the return value?

Reply:
> Do you think we should also allow `_` in addition to `__return__` for the return value?

I feel less strongly about this, either way. Perhaps just `__return__`. Programmatic interfaces don't need the same niceties, because people can always build a layer on top of it.